sklearn/utils/arrayfuncs.pyx

```diff
@@ -18,36 +18,27 @@ ctypedef np.float64_t DOUBLE
 np.import_array()
 
 
-def min_pos(np.ndarray X):
-   """
-   Find the minimum value of an array over positive values
-
-   Returns a huge value if none of the values are positive
-   """
-   if X.dtype.name == 'float32':
-      return _float_min_pos(<float *> X.data, X.size)
-   elif X.dtype.name == 'float64':
-      return _double_min_pos(<double *> X.data, X.size)
-   else:
-      raise ValueError('Unsupported dtype for array X')
-
-
-cdef float _float_min_pos(float *X, Py_ssize_t size):
-   cdef Py_ssize_t i
-   cdef float min_val = DBL_MAX
-   for i in range(size):
-      if 0. < X[i] < min_val:
-         min_val = X[i]
-   return min_val
-
-
-cdef double _double_min_pos(double *X, Py_ssize_t size):
-   cdef Py_ssize_t i
-   cdef np.float64_t min_val = FLT_MAX
-   for i in range(size):
-      if 0. < X[i] < min_val:
-         min_val = X[i]
-   return min_val
+def min_pos(np.ndarray X): 
+   """ 
+   Find the minimum value of an array over positive values 
+ 
+   Returns a huge value if none of the values are positive 
```

Comment:
I see you are copying this over from before, but this can be slightly improved:

```suggestion
   Returns maximum representable value of the input dtype if none of the values are positive 
```

Reply:
Agreed, done.